src/os/data/layersyncdescriptor.js

```diff
@@ -109,6 +120,45 @@ os.data.LayerSyncDescriptor.prototype.disposeInternal = function() {
 };
 
 
+/**
+ * @inheritDoc
+ */
+os.data.LayerSyncDescriptor.prototype.getMappings = function() {
+  return this.mappings;
+};
+
+
+/**
+ * @inheritDoc
+ */
+os.data.LayerSyncDescriptor.prototype.setMappings = function(value) {
+  this.mappings = value;
+  const dm = os.data.DataManager.getInstance();
+  dm.updateDescriptor(this, this);
+  dm.persistDescriptors();
+};
+
+/**
+ * @inheritDoc
+ */
+os.data.LayerSyncDescriptor.prototype.updateMappings = function(layer) {
+  const dm = os.data.DataManager.getInstance();
+  dm.updateDescriptor(this, this);
+  dm.persistDescriptors();
+
+  // Delete the layer, then prompt the descriptor to make new layers
```

Comment:
It's a little odd that this implementation only removes the layer, and leaves it to extending classes to add the layer back. This should be a complete implementation on its own. Can we recreate the layer here? It would be nice if descriptors like OGC and Arc didn't need to add their own mapping logic and just got it for free from `LayerSyncDescriptor`.